corehq/apps/reports/filters/api.py

```diff
@@ -247,3 +249,30 @@ def paginate_options(data_sources, query, start, size):
         size -= len(objects)  # how many more do we need for this page?
         options.extend(objects)
     return total, options
+
+
+class DeviceLogFilter(LoginAndDomainMixin, JSONResponseMixin, View):
+    field = None
+
+    def get(self, request, domain):
+        q = self.request.GET.get('q', None)
+        field_filter = {self.field + "__startswith": q}
+        values = (
+            DeviceReportEntry.objects
+            .filter(domain=domain)
+            .filter(**field_filter)
+            .distinct(self.field)
+            .values_list(self.field, flat=True)
+        )[:10]
```

Comment:
django doesn't smartly add `limit 10` right? specifying that could help with the number of objects loaded into memory